src/scope/scope.js

```diff
@@ -400,6 +400,35 @@ export default class Scope {
       });
   }
 
+  reset({ bitId, consumer }: { bitId: BitId, consumer?: Consumer}): Promise<consumerComponent> {
+    if (!bitId.isLocal(this.name)) {
+      return Promise.reject('you can not reset a remote component');
+    }
+    return this.sources.get(bitId)
+      .then((component) => {
+        if (!component) throw new ComponentNotFound(bitId.toString());
+        const allVersions = component.listVersions();
+        if (allVersions.length > 1) {
+          const lastVersion = component.latest();
+          bitId.version = lastVersion.toString();
+          return consumer.removeFromComponents(bitId, true).then(() => {
+            bitId.version = (lastVersion - 1).toString();
+            return this.get(bitId).then((consumerComponent) => {
+              const ref = component.versions[lastVersion];
+              return this.objects.remove(ref).then(() => { // todo: remove also all deps of that ref
```

Comment:
> // todo: remove also all deps of that ref

Do you mean to the impl and spec files sources for example?
I'm not sure you can just delete them because they can belong to another version.